app/models/piece.rb

```diff
@@ -32,6 +41,24 @@ def move_to!(x, y)
     raise ArgumentError, "Can't move piece" unless update_attributes(x_coordinate: x, y_coordinate: y)
   end
 
+  # testing potential moves - move will not be saved
+  def move_to(x, y)
+    # check to see if there is a piece in the location it`s moving to.
+    test_destination_piece = Piece.find_by(game: game, x_coordinate: x, y_coordinate: y, captured: false)
+
+    # Second, if there is a piece there, and it's the opposing color, remove the piece from the board.
+    if test_destination_piece
+      # Third, if the piece is there and it's the same color the move should fail - it should either raise an error message or do nothing.
+      raise ArgumentError, "Can't take your own piece!" if test_destination_piece.color == color
+
+      # You could set the piece's x/y coordinates to nil
+      test_destination_piece.update_attributes(x_coordinate: nil, y_coordinate: nil, captured: true)
```

Comment:
what's the purpose of this line?
In my opinion you can just remove this line and do `... update_attributes(x_coordinate: x, y_coordinate: y, captures: true)` on line 59